CurrentChargingTest.cpp

```diff
@@ -0,0 +1,20 @@
+#include "CurrentCharging.h"
+#include <cassert>
+
+int main()
+{
+    vector<int> periodicSamples = {1, 3, 5, 3, 4, 2, 27, 29, 28, 30, 10, 11, 12, 53, 51, 52};
+    CCurrentReport obj(periodicSamples);
+    obj.generateContinuesCurrentRange();
+    obj.consoleReport();
+    assert(obj.validateRangeandReadings(1, 5, 6) == true);
```

Comment:
These asserts don't talk about the way in which ranges and counts are reported. It's hard to know what to expect from the implementation. All tests of the implementation need to be manual, making development inefficient.
Assert the expected ranges.